chia/wallet/wallet_node.py

```diff
@@ -1003,30 +1003,33 @@ async def get_timestamp_for_height_from_peer(self, height: uint32, peer: WSChiaC
         a transaction block
         """
         cache = self.get_cache_for_peer(peer)
-        last_tx_block = None
         request_height: int = height
         while request_height >= 0:
             cached_timestamp = cache.get_height_timestamp(uint32(request_height))
             if cached_timestamp is not None:
                 return cached_timestamp
-            response: Optional[List[HeaderBlock]] = await request_header_blocks(
-                peer, uint32(request_height), uint32(request_height)
-            )
-            if response is not None and len(response) > 0:
-                cache.add_to_blocks(response[0])
-                if response[0].is_transaction_block:
-                    last_tx_block = response[0]
+            block = cache.get_block(uint32(request_height))
+            if block is None:
+                self.log.debug(f"get_timestamp_for_height_from_peer cache miss for height {request_height}")
+                response: Optional[List[HeaderBlock]] = await request_header_blocks(
+                    peer, uint32(request_height), uint32(request_height)
+                )
+                if response is not None and len(response) > 0:
+                    self.log.debug(f"get_timestamp_for_height_from_peer add to cache for height {request_height}")
+                    cache.add_to_blocks(response[0])
```

Comment:
if the only thing we use this cache for is to get timestamps, we don't technically need to cache non-tx blocks. But it looks like it works